pkg/reconciler/namespace/namespace.go

```diff
@@ -172,6 +175,14 @@ func (c *Controller) assignCluster(ctx context.Context, ns *corev1.Namespace) er
 			klog.V(2).InfoS("assignCluster: excluding cluster with different metadata.clusterName", "ns.clusterName", ns.ClusterName, "check", allClusters[i].ClusterName)
 			continue
 		}
+		if allClusters[i].Spec.Unschedulable {
+			klog.V(2).InfoS("assignCluster: excluding unschedulable cluster", "metadata.name", allClusters[i].Name)
+			continue
+		}
+		if evictAfter := allClusters[i].Spec.EvictAfter; evictAfter != nil && evictAfter.Time.Before(now()) {
+			klog.V(2).InfoS("assignCluster: excluding evicted cluster", "metadata.name", allClusters[i].Name)
```

Comment:
nit: Is there a better term for 'cluster whose contents are marked for eviction' than 'evicted cluster'?

Reply:
Maybe. In the meantime, using more words to add clarity.